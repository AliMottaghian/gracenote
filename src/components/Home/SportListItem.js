import React from "react";
import PropTypes from "prop-types";
import { Row, Col, Button } from "antd";
import { UnorderedListOutlined } from "@ant-design/icons";

const SportListItem = ({ history, name, id }) => {
  const showDetails = () => {
    history.push(`/detail/${id}`);
  };
  return (
    <Row>
      <Col xs={18} md={20}>
        <strong style={{ lineHeight: 2 }}>{name}</strong>
      </Col>
      <Col xs={6} md={4}>
        <Button
          type="default"
          icon={<UnorderedListOutlined />}
          onClick={showDetails}
        >
          Details
        </Button>
      </Col>
    </Row>
  );
};
SportListItem.propTypes = {
  history: PropTypes.object.isRequired,
  name: PropTypes.string,
  id: PropTypes.number.isRequired,
};

export default SportListItem;
