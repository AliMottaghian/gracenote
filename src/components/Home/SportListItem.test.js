import SportListItem from "./SportListItem";

describe("<SportListItem />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SportListItem name="Soccer" id={1} history={{ push: () => {} }} />
    );
  });

  it("should render App component without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render the name in Strong tag", () => {
    const strong = wrapper.find("strong");
    expect(strong.text()).toBe("Soccer");
  });
});
