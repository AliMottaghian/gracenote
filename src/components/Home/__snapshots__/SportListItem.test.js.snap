// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SportListItem /> should render App component without crashing 1`] = `
<Row>
  <Col
    md={20}
    xs={18}
  >
    <strong
      style={
        Object {
          "lineHeight": 2,
        }
      }
    >
      Soccer
    </strong>
  </Col>
  <Col
    md={4}
    xs={6}
  >
    <Button
      icon={<ForwardRef(UnorderedListOutlined) />}
      onClick={[Function]}
      type="default"
    >
      Details
    </Button>
  </Col>
</Row>
`;
